src/ServiceManagerBase.sol

```diff
@@ -135,6 +140,77 @@ abstract contract ServiceManagerBase is ServiceManagerBaseStorage {
         _setRewardsInitiator(newRewardsInitiator);
     }
 
+    function migrateAndCreateOperatorSetIds(uint32[] memory operatorSetsToCreate) external onlyOwner{
+        _migrateAndCreateOperatorSetIds(operatorSetsToCreate);
+    }
+
+    function migrateToOperatorSets(uint32[][] memory operatorSetIds, address[] memory operators) external onlyOwner {
+        require(!migrationFinalized, "SerivceManager: Migration Already Finalized");
+        _migrateToOperatorSets(operatorSetIds, operators);
+    }
+
+    function finalizeMigration() external onlyOwner{
+        require(!migrationFinalized, "SerivceManager: Migration Already Finalized");
+        migrationFinalized = true;
+    }
+
+    function _migrateAndCreateOperatorSetIds(uint32[] memory operatorSetIdsToCreate) internal {
+        _avsDirectory.becomeOperatorSetAVS();
+        IAVSDirectory(address(_avsDirectory)).createOperatorSets(operatorSetIdsToCreate);
+    }
+
+    function _migrateToOperatorSets(uint32[][] memory operatorSetIds, address[] memory operators) internal {
+        IAVSDirectory(address(_avsDirectory)).migrateOperatorsToOperatorSets(operators, operatorSetIds);
```

Comment:
are we not checking that the provided operators are correct?

Reply:
there's a race condition here where operators deregister for a certain quorum inbetween when the view function is called and migrate is called

Reply:
We can and just skip. May be cleaner

Reply:
Regarding race condition, we can also prevent m2 deregistrations in AVSDirectory

Reply:
yeah we should block m2 functions in core  
`if (isOperatorSetAVS[avs]==true)`

Reply:
added additional checks [d64f4af](https://github.com/Layr-Labs/eigenlayer-middleware/pull/288/commits/d64f4af465201d66e2c581e42f28b32764a82381)

Reply:
added pr here to block m2 dereg: https://github.com/Layr-Labs/eigenlayer-contracts/pull/688